sumologic/resource_sumologic_slo.go

```diff
@@ -0,0 +1,416 @@
+const sloContentType = "slo"
+
+func resourceSumologicSLO() *schema.Resource {
+
+	aggrRegex := regexp.MustCompile(sloAggregationRegexString)
+	windowRegex := regexp.MustCompile(sloAggregationWindowRegexString)
+
+	queryGroupElemSchema := &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"row_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"query": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"use_row_count": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"field": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+
+	return &schema.Resource{
+		Create: resourceSumologicSLOCreate,
+		Read:   resourceSumologicSLORead,
+		Update: resourceSumologicSLOUpdate,
+		Delete: resourceSumologicSLODelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"version": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"created_by": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"modified_at": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"modified_by": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"parent_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"is_system": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+			},
+			//"content_type": {
+			//	Type:     schema.TypeString,
+			//	Optional: true,
+			//	Default:  "Slo",
+			//	ExactlyOneOf: []string{
+			//		"Slo",
+			//	},
+			//},
+			//"type": {
+			//	Type:     schema.TypeString,
+			//	Optional: true,
+			//	Default:  "SlosLibraryFolder",
+			//	ExactlyOneOf: []string{
+			//		"SlosLibrarySlo",
+			//		"SlosLibrarySloUpdate",
+			//	},
+			//},
+			"signal_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"compliance": {
+				Type:     schema.TypeList,
+				Required: true,
+				//MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"compliance_type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"Rolling",
+								"Calendar",
+							}, false),
+						},
+						"target": {
+							Type:         schema.TypeFloat,
+							Required:     true,
+							ValidateFunc: validation.FloatBetween(0, 100),
+						},
+						"timezone": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"size": {
```

Comment:
How would this work for Calendar compliance? How would user set up weekly with week begins on Monday?